client/worker.go

```diff
@@ -49,22 +49,20 @@ const (
 // and data corruption. In case a caller needs to export multiple requests at the same time, it should
 // use one worker per request.
 type worker struct {
-	closeCh     chan struct{}
 	client      *http.Client
 	accessToken string
 	endpoint    string
 	gzipWriter  *gzip.Writer
-	maxRetries  uint
+	pauseCB     func(time.Duration)
 }
 
-func newWorker(closeCh chan struct{}, client *http.Client, endpoint string, accessToken string, maxRetries uint) *worker {
+func newWorker(pauseCB func(time.Duration), client *http.Client, endpoint string, accessToken string) *worker {
```

Comment:
The callback approach looks somewhat more complicated than I think it needs to be.

What if instead `Client.Export()` checks the error returned by `worker.export()` and initiates the pause? In that case there will be no need for this callback.

Reply:
Updated